scripts/migrations/shared.sh

```diff
@@ -0,0 +1,57 @@
+function update_json_if_not_empty() {
+  local variable_name="$1"
+  local json_key="$2"
+  local value="$3"
+  local is_json_field="$4"
+
+  if [ "$value" ]; then
+    if [[ "$is_json_field" == "false" || -z "$is_json_field" ]]; then
+      contents="$(jq --arg $variable_name "$value" "$json_key = \"$value\"" $ROOT_PATH/euclid.json)" &&
+        echo -E "${contents}" >$ROOT_PATH/euclid.json
+    else
+      contents="$(jq --argjson $variable_name "$value" "$json_key = $value" $ROOT_PATH/euclid.json)" &&
+        echo -E "${contents}" >$ROOT_PATH/euclid.json
+    fi
+
+  fi
+}
+
+function migrate_fields(){
```

Comment:
Won't you need a separate migrate_fields function for every version migration? Each migration is supposed to run in sequence so that a user can update from an old version, through several newer versions, to the latest version. I don't think this can safely be shared.

Reply:
This name isn't so good, In this case, we're just copying the previous values to the new euclid.json file. In the migration process, we clone the updated euclid.json, and copy the current one to an older-euclid.json, and then we call this function to copy previous values to the new euclid.json file, so even sequentially it's copying